aip/general/0148.md

```diff
@@ -137,6 +128,25 @@ used by server-side policies, such as IAM conditions. Annotations exist to
 allow client tools to store their own state information without requiring a
 database.
 
+### Well known string fields
+
+#### IP address
+
+A field that represents an IP address **must** comply with the following:
+
+* use type `string`
+* use the name `ip_address` or end with the suffix `_ip_address` e.g.
+  `resolved_ip_address`
+* specify the IP address version format via one of the supported formats `IPV4`,
+  `IPV6`, or if it can be either, `IPV4_OR_IPV6` (see [AIP-202][aip-202])
+
+#### uid
+
+The output only `string uid` field refers to a system-assigned, unique
+identifier for a resource. When provided, this field **should** be a [UUID4][]
+and **must** specify this format via the `UUID4` format extension (see
```

Comment:
The combination of must and should is a bit odd here... because that would suggest it's "just about okay" to use a value that *isn't* a UUID-4, but *does* specify the format via the `UUID4` format extension.

Perhaps

> When provided, this field **should** be a [UUID4][], and (if it is) **must** specify this format via the `UUID4` format extension

Reply:
(Or just turn them both into must?)

Reply:
I see what you mean. I was going for what was done in `request_id` and not trying to tighten other guidance, just ensure that when applicable the format extension is used. I see how that is confusing though.

I can change it to a "must", but I have no way to verify that existing examples are compliant. I know that ccfe uses uuid for this field when involved, but existing cases outside of ccfe I have no way to verify.

I'm not opposed to tightening it though, I don't see a reason for it to be anything else. I will do so and bring it up at the next meeting to discuss if it should be reverted.